### PR TITLE
Add UCenter user authentication method.

### PR DESCRIPTION
**Pull request by AxelPanda**
UCenter is a famous and popular PHP user center developed by Comsenz (http://www.comsenz.com). UCenter usually works together with `Discuz!` or `Discuz! X`, which is the most popular PHP forum software among Chinese communities and users. Here is my solution for integrating UCenter with ownCloud, which allows users who are already registered in UCenter to sign in ownCloud with their exactly the same account.

Usage: edit `config/config.php` and adding the following:
```
"user_backends" => array (
    0 => array (
            "class"     => "OC_User_UC",
            "arguments" => array (
                              0 => 'myUcApi'
                              ),
    ),
),
```

### CODE REVIEW

**Comment by karlitschek** [comment]
Thanks or your contribution. Can test this at the moment but looks fine to me. Unfortunately we are in feature freeze for 8.2 at the moment. This is probably fine to be merged for 9.0 in about 4 weeks. 
Some testers/reviewers would be good.

File: user_external/lib/uc_client/model/base.php [hunk 1]
```diff
@@ -0,0 +1,254 @@
+<?php
+
+/*
+	[UCenter] (C)2001-2099 Comsenz Inc.
+	This is NOT a freeware, use is subject to license terms
```
Comment (LukasReschke) [inline user_external/lib/uc_client/model/base.php]: :-1:

**Comment by LukasReschke** [comment]
Unclear licensing, source code is not using prepared statements, dynamically evaluating code, no unit tests. I'm quite reluctant to add this change to the core product.

You're more than welcome to create a dedicated application for this use-case. But I'm strongly against merging this into the user_external which will be shipped to all users.

:-1: from my side.

**Comment by LukasReschke** [comment]
See https://github.com/owncloud/user_cas for an example of an ownCloud backend in a dedicated application.

**Comment by AxelPanda** [comment]
Thanks, I'll try to build it as a dedicated application.

**Pull request closed without merging**